src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -429,19 +430,22 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Target Name="_GetRestoreProjectStyle"
     DependsOnTargets="_GetProjectJsonPath;CollectPackageReferences"
     Returns="$(RestoreProjectStyle);$(PackageReferenceCompatibleProjectStyle)">
-    <!-- This may be overridden by setting RestoreProjectStyle in the project. -->
     <PropertyGroup>
-      <!-- If any PackageReferences exist treat it as PackageReference. This has priority over project.json. -->
-      <RestoreProjectStyle Condition=" '$(RestoreProjectStyle)' == '' AND @(PackageReference) != '' ">PackageReference</RestoreProjectStyle>
-      <!-- If this is not a PackageReference project check if project.json or projectName.project.json exists. -->
-      <RestoreProjectStyle Condition=" '$(RestoreProjectStyle)' == '' AND '$(_CurrentProjectJsonPath)' != '' ">ProjectJson</RestoreProjectStyle>
-      <!-- If this is not a PackageReference or ProjectJson project check if packages.config or packages.ProjectName.config exists -->
-      <RestoreProjectStyle Condition=" '$(RestoreProjectStyle)' == '' AND (Exists('$(MSBuildProjectDirectory)\packages.config') Or Exists('$(MSBuildProjectDirectory)\packages.$(MSBuildProjectName).config'))">PackagesConfig</RestoreProjectStyle>
-      <!-- This project is either a packages.config project or one that does not use NuGet at all. -->
-      <RestoreProjectStyle Condition=" '$(RestoreProjectStyle)' == '' ">Unknown</RestoreProjectStyle>
+      <_HasPackageReferenceItems Condition="@(PackageReference->Count()) > 0">true</_HasPackageReferenceItems>
     </PropertyGroup>
+
+    <GetRestoreProjectStyleTask
+      HasPackageReferenceItems="$(_HasPackageReferenceItems)"
+      MSBuildProjectDirectory="$(MSBuildProjectDirectory)"
+      MSBuildProjectName="$(MSBuildProjectName)"
+      ProjectJsonPath="$(_CurrentProjectJsonPath)"
+      RestoreProjectStyle="$(RestoreProjectStyle)">
+        <Output TaskParameter="ProjectStyle" PropertyName="RestoreProjectStyle" />
+        <Output TaskParameter="IsPackageReferenceCompatibleProjectStyle" PropertyName="PackageReferenceCompatibleProjectStyle" />
+    </GetRestoreProjectStyleTask>
+
     <PropertyGroup>
-      <PackageReferenceCompatibleProjectStyle Condition="  '$(RestoreProjectStyle)' == 'PackageReference' OR '$(RestoreProjectStyle)' == 'DotnetToolReference' ">true</PackageReferenceCompatibleProjectStyle>
+      <_HasPackageReferenceItems />
```

Comment:
Can you explain this?

Reply:
I want to pass a `Boolean` to the task, but I need to get the info from whether or not there are any PackageReference items.  MSBuild has an item function `Count()` which returns an `int.  The only way I could find to translate the `int` to a `bool` was to create a property and use a condition.

Another option would be to pass the total number of `PackageReference` items and assume if its greater than one then its `PackageReference` style.  But I thought a `bool` was more concise.

Reply:
I understand that and it makes sense, but why does it need to be declared here?
I guess I'm missing something obvious :)

Reply:
This particular line is just clearing the property value since its only used temporarily to execute the task